jvm/middleware/src/main/kotlin/io/spine/logging/jvm/MiddlemanApi.kt

```diff
@@ -0,0 +1,1058 @@
+     * ```
+     * currentTimestampNanos >= lastTimestampNanos + unit.toNanos(n)
+     * ```
+     * where `currentTimestampNanos` is the timestamp of the current log statement and
+     * `lastTimestampNanos` is a time stamp of the last log statement that was emitted.
+     *
+     * The effect of this is that when logging invocation is relatively infrequent,
+     * the period between emitted log statements can be higher than the specified duration.
+     * For example, if the following log statement were called every 600ms:
+     *
+     * ```kotlin
+     * logger.atFine().atMostEvery(2, SECONDS).log(...)
+     * ```
+     *
+     * logging would occur after `0s`, `2.4s` and `4.8s` (not `4.2s`),
+     * giving an effective duration of `2.4s` between log statements over time.
+     *
+     * Providing a zero-length duration (i.e., `n == 0`) disables rate limiting and
+     * makes this method an effective no-op.
+     *
+     * ### Granularity
+     *
+     * Because the implementation of this feature relies on a nanosecond timestamp
+     * provided by the backend, the actual granularity of the underlying clock used
+     * may vary, and it is possible to specify a time period smaller than
+     * the smallest visible time increment.
+     *
+     * If this occurs, then the effective rate limit applied to the log statement
+     * will be the smallest available time increment.
+     *
+     * For example, if the system clock granularity is 1 millisecond, and
+     * a log statement is called with `atMostEvery(700, MICROSECONDS)`,
+     * the effective rate of logging (even averaged over long periods)
+     * could never be more than once every millisecond.
+     *
+     * ### Notes
+     *
+     * If *multiple rate limiters* are used for a single log statement, that
+     * log statement will only be emitted once all rate limiters have reached their threshold,
+     * and when a log statement is emitted all the rate limiters are reset.
+     *
+     * So even if the rate limit duration has expired, it does not mean that logging will occur.
+     *
+     * When rate limiting is active, a "skipped" count is added to log statements
+     * to indicate how many logs were disallowed since the last log statement was emitted.
+     *
+     * If this method is called multiple times for a single log statement,
+     * the last invocation will take precedence.
+     *
+     * @param n The minimum number of time units between emitted log statements
+     * @param unit The time unit for the duration
+     * @throws IllegalArgumentException if [n] is negative.
+     */
+    public fun atMostEvery(n: Int, unit: TimeUnit): API
+
+    /**
+     * Returns true if logging is enabled at the level implied for this API,
+     * according to the current logger backend.
+     *
+     * For example:
+     *
+     * ```kotlin
+     * if (logger.atFine().isEnabled()) {
+     *   // Do non-trivial argument processing
+     *   logger.atFine().log("Message: %s", value)
```

Comment:
I understand this `%s` is a leftover rudiment from Java's way of printing values, but I wouldn't expect this in Kotlin.

Leaving this comment so that we don't forget to revise the sample code coming from the original Flogger documentation, in the future.